src/UndoRedo/Changes/TreeItemAddChange.ts

```diff
@@ -92,25 +97,23 @@ class TreeItemAddChange extends Change {
    * @param {object} j -The j treeItem
    * @param {object} context - The context treeItem
    */
-  fromJSON(j, context) {
-    const treeItem = Registry.constructClass(j.treeItem.type)
+  fromJSON(j: Record<any, any>, context: Record<any, any>) {
+    const treeItem = <TreeItem>Registry.constructClass(j.treeItem.type)
     if (!treeItem) {
       console.warn('resolvePath is unable to construct', j.treeItem)
       return
     }
     this.name = j.name
     this.treeItem = treeItem
-    this.treeItem.addRef(this)
 
     this.treeItem.fromJSON(j.treeItem, context)
-    this.treeItemIndex = this.owner.addChild(this.treeItem, false, false)
+    this.treeItemIndex = this.owner.getChildIndex(this.owner.addChild(this.treeItem))
   }
```

Comment:
curious about this line. I didn't realize 'addChild' returned the child , and not the index.

Reply:
Yeah, I found this in a few areas.